pydicom/pixel_data_handlers/util.py

```diff
@@ -335,19 +336,20 @@ def apply_voi_lut(arr, ds, index=0):
         elem = ds['WindowWidth']
         width = elem.value[index] if elem.VM > 1 else elem.value
 
-        # The output range depends on whether or not a rescale operation has
-        #   been applied - if not then the range is the available bit depth
-        if ds.PixelRepresentation == 1:
-            # Signed
-            y_min = -2**(ds.BitsStored - 1)
-            y_max = 2**(ds.BitsStored - 1) - 1
-        else:
+        # The output range depends on whether or not a modality LUT or rescale
+        #   operation has been applied - if not then the range is the available
+        #   bit depth - see PS3.3 C.11.1.1.1
+        if ds.PixelRepresentation == 0 or 'ModalityLUTSequence' in ds:
             # Unsigned
             y_min = 0
             y_max = 2**ds.BitsStored - 1
```

Comment:
In the case of a Modality LUT sequence, shouldn't that be rather:
`2**ds.ModalityLUTSequence[0].LUTDescriptor[2] - 1` ?

Reply:
Whoops, yep

Reply:
Really done this time, probably